pkg/certificate/types.go

```diff
@@ -153,3 +156,16 @@ type MRCEventBroker interface {
 	// an error if the subscription goes awry.
 	Watch(context.Context) (<-chan MRCEvent, error)
 }
+
+// UpdateMRCStatus updates the status of an MRC to the desired state
+type UpdateMRCStatus func(ctx context.Context, mrc *v1alpha2.MeshRootCertificate) error
```

Comment:
Is there a strong reason to type alias these methods?

The CheckMRCStatus doesn't make any API calls, so I imagine there will only be one implementation. I'd recommend we just make that a pointer receiver.

The UpdateMRCStatus is analogous to UpdateMeshRootCertificateStatus, so that can just be embedded

Reply:
From my understanding, users will pass in an arbitrary `UpdateMRCStatus` function for the reconciler to call later, thus the type alias

Reply:
so there's 2 methods defined. To me that means it should be an interface. 

["Interfaces are named collections of method signatures."](https://gobyexample.com/interfaces)

That would further remove the need for a type alias. For simplicity, and to avoid duplication, we should define the interface such that the compute.Client/k8s.Controller adheres to that interface

Reply:
These aren't "methods" though in the truest sense. These functions don't need receivers, and usually won't have receivers